packages/backend/src/clients/Slack/Slackbot.ts

```diff
@@ -1,38 +1,165 @@
+                    },
+                },
+            ],
+        },
+    };
+    await client.chat
+        .unfurl({
+            ts: event.message_ts,
+            channel: event.channel,
+            unfurls,
+        })
+        .catch((er: any) =>
+            Logger.error(`Unable to unfurl slack URL ${url}: ${error} `),
+        );
+};
+
+type SlackServiceDependencies = {
+    slackAuthenticationModel: SlackAuthenticationModel;
+};
+
+export class SlackService {
+    slackAuthenticationModel: SlackAuthenticationModel;
+
+    constructor({ slackAuthenticationModel }: SlackServiceDependencies) {
+        this.slackAuthenticationModel = slackAuthenticationModel;
+        this.start();
+    }
+
+    async start() {
+        if (process.env.SLACK_APP_TOKEN) {
+            try {
+                const slackReceiver = new ExpressReceiver({
+                    ...slackOptions,
+                    installationStore: {
+                        storeInstallation: (i) =>
+                            this.slackAuthenticationModel.createInstallation(i),
+                        fetchInstallation: (i) =>
+                            this.slackAuthenticationModel.getInstallation(i),
+                        deleteInstallation: (i) =>
+                            this.slackAuthenticationModel.deleteInstallation(i),
+                    },
+                    router: apiV1Router,
+                });
+
+                await slackReceiver.start(parseInt('4001', 10));
 
-const slackReceiver = new ExpressReceiver({
-    ...slackOptions,
-    router: apiV1Router,
-});
-
-export const startSlackBot = async () => {
-    if (process.env.SLACK_APP_TOKEN) {
-        try {
-            await slackReceiver.start(parseInt('3001', 10));
-
-            const app = new App({
-                ...slackOptions,
-                logLevel: LogLevel.INFO,
-                port: parseInt(process.env.SLACK_PORT || '4000', 10),
-                socketMode: true,
-                appToken: process.env.SLACK_APP_TOKEN,
-            });
-
-            app.event('link_shared', async (message: any) => {
-                const { event, client, context } = message;
-                await slackService.unfurl(event, client, context);
-            });
-
-            await app.start();
-
-            console.debug('Slack app is running');
-        } catch (e: unknown) {
-            console.error(`Unable to start Slack app ${e}`);
+                const app = new App({
+                    ...slackOptions,
+                    installationStore: {
+                        storeInstallation: (i) =>
+                            this.slackAuthenticationModel.createInstallation(i),
+                        fetchInstallation: (i) =>
+                            this.slackAuthenticationModel.getInstallation(i),
+                    },
+                    logLevel: LogLevel.INFO,
+                    port: parseInt(process.env.SLACK_PORT || '4000', 10),
```

Comment:
not sure if that makes sense but I think we should start using weird port number defaults because 4000, 3000, 8080 are very common to be used from developers and docker might not be able to start because of that